### DISCUSSION TITLE
[app.bsky.feed.searchPosts] Use the public firehose or Jetstream to get all posts

### DISCUSSION THREAD

**Discussion by sabatale** [bluesky-social/atproto#3005] [How to do it?]
Great work so far!

We had a quick question regarding the HTTP docs: [https://docs.bsky.app/docs/api/app-bsky-feed-search-posts.](https://docs.bsky.app/docs/api/app-bsky-feed-search-posts.)

This does not work: `use the public firehose or Jetstream to get all posts`
[https://public.api.bsky.app/xrpc/app.bsky.feed.searchPosts?q=from%3Awashingtonpost.com+since%3A2023-01-02+until%3A2023-06-02](https://public.api.bsky.app/xrpc/app.bsky.feed.searchPosts?q=from%3Awashingtonpost.com+since%3A2023-01-02+until%3A2023-06-02)

But this does:
[https://public.api.bsky.app/xrpc/app.bsky.feed.searchPosts?q=washingtonpost.com+since%3A2023-01-02+until%3A2023-06-02](https://public.api.bsky.app/xrpc/app.bsky.feed.searchPosts?q=washingtonpost.com+since%3A2023-01-02+until%3A2023-06-02)

The difference seems to be the additional filter in query `from:`. But the docs don't seem to mention it.

What is the right way to proceed? Use `searchposts` for general keywords, and `getAuthorFeed` directly for an actor?

If that's the latter, there is no mention of date filters either.

Thanks!

**Comment by bnewbold** [comment]
what is it that you are trying to achieve?

**Reply by sabatale** [reply]
Filter by dates on any type of query (`q=`).

**Comment by mary-ext** [comment]
Resources around search syntax is available here: https://bsky.social/about/blog/05-31-2024-search

**Reply by sabatale** [reply]
That's what we've been doing. But as soon as we add `since` or `until` to a query with `from:` - it fails.

[https://public.api.bsky.app/xrpc/app.bsky.feed.searchPosts?q=from%3Awashingtonpost.com&since=2023-01-02T23:00:00.000Z](https://public.api.bsky.app/xrpc/app.bsky.feed.searchPosts?q=from%3Awashingtonpost.com&since=2023-01-02T23:00:00.000Z)
